src/Middleware/UnauthorizedHandler/RedirectHandler.php

```diff
@@ -92,8 +92,8 @@ protected function checkException(Exception $exception, array $exceptions)
     protected function getUrl(ServerRequestInterface $request, array $options)
     {
         $url = $options['url'];
-        if ($options['queryParam'] !== null) {
-            $query = urlencode($options['queryParam']) . '=' . urlencode($request->getUri());
+        if ($options['queryParam'] !== null && $request->getMethod() === 'GET') {
```

Comment:
I also found redirects on non-GET which is something to be avoided as those URLs could have side-effects or generate 405 errors.